### ISSUE TITLE
JavaScript Support?

### ISSUE THREAD

**Issue by alberto467** [cardigann/cardigann#281]
I've been writing a definition for ExtraTorrent, it worked pretty well, until they updated they're website.
Now, the table with the torrents is not sent to the user in html, it is 'encrypted' and encoded in base64 on their end and then it is de-obfuscated by the JavaScript in the browser into html.
There are not too many reasons for why they would do this, i would guess blocking auto scrapers was their intent, and it worked. What i'm worried about is that more and more indexers will start doing this...
I know this is not something easy to implement and it's going to be a future thing, but we might HAVE to do this if we want to keep cardigann alive in the future.

**Comment by alberto467** [comment]
UPDATE: i just checked and it looks like they disabled the obfuscation, so we're good for now.

**Comment by kaso17** [comment]
Probably related: https://torrentfreak.com/extratorrent-takes-on-leeching-proxy-sites-161219/

**Comment by lxt2** [comment]
I started work on https://github.com/cardigann/go-cloudflare-scraper a while back, it includes a JS interpreter. Been meaning to get some time to finish it.

**Comment by qwertyhup** [comment]
Yes, the html on extratorrent.cc is still encrypted. Please tell me how to decrypt the code manually.

**Comment by thebluepotato** [comment]
Apparently they've had some success : https://github.com/ChrisAlderson/extratorrent-api/blob/master/extratorrent-api.js It's a node.js script.

**Comment by kaso17** [comment]
Before the weekend I had a working decrypter for their ajax_checker key retrieval obfuscation. Now they completely changed their method again to something closer to the extratorrent-api.js solution but I don't thing it will work at this time. Seems like they're changing obfuscation methods from time to time. Currently the easiest solution seems to be using a full blown headless browser to workaround the problem. At some point they probably would start using CAPTCHAS to prevent automated access.
As i don't use public trackers anyway I'm not motivated to invest more time into this for now.
